src/pkg/services/m365/api/drive_pager.go

```diff
@@ -204,20 +205,20 @@ func (c Drives) EnumerateDriveItemsDelta(
 	driveID string,
 	prevDeltaLink string,
 	cc CallConfig,
-) NextPageResulter[models.DriveItemable] {
+) pagers.NextPageResulter[models.DriveItemable] {
 	deltaPager := c.newDriveItemDeltaPager(
 		driveID,
 		prevDeltaLink,
 		cc)
 
-	npr := &nextPageResults[models.DriveItemable]{
-		pages: make(chan nextPage[models.DriveItemable]),
+	npr := &pagers.NextPageResults[models.DriveItemable]{
+		Pages: make(chan pagers.NextPage[models.DriveItemable]),
 	}
```

Comment:
this is the only place `Pages` is accessed so it seems reasonable to just make an initializer function for the struct so we can keep the field unexported